src/illumidesk/authenticators/authenticator.py

```diff
@@ -153,6 +156,54 @@ async def authenticate(self, handler, data=None):
                 },
             }
 
+    async def post_auth_hook(self, authenticator, handler, authentication):
+        """
+            Calls the microservice to setup up a new course in case it does not exist.          
+            The data needed is received from auth_State
+        """
```

Comment:
I believe we need to document the parameters for this function definition even though its documented upstream in JupyterHub's documentation. We should probably re-emphasize that we need to return the authentication dict.